xarray/backends/api.py

```diff
@@ -202,6 +202,21 @@ def check_attr(name, value, valid_types):
                 f"{', '.join([vtype.__name__ for vtype in valid_types])}"
             )
 
+        if isinstance(value, bytes) and engine == "h5netcdf":
+            try:
+                value.decode("utf-8")
+            except UnicodeDecodeError as e:
+                raise ValueError(
+                    f"Invalid value provided for attribute '{name!r}': {value!r}. "
+                    "Only binary data derived from UTF-8 encoded strings is allowed."
```

Comment:
Is this only applicable to `h5netcdf`? shall we mention that in the string? We could also mention `engine="netCDF4" as an alternative (assuming that works).